internal/common/armadaerrors/errors_test.go

```diff
@@ -0,0 +1,155 @@
+package armadaerrors
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	grpc_middleware "github.com/grpc-ecosystem/go-grpc-middleware"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+
+	"github.com/G-Research/armada/internal/common/requestid"
+)
+
+func TestCodeFromError(t *testing.T) {
+	tests := map[string]struct {
```

Comment:
Does this actually test where we skip if the error is already a status error?

Reply:
We do now :)